client/components/pages/upload/FileTile.vue

```diff
@@ -0,0 +1,108 @@
+<script setup lang="ts">
+const { file, status } = defineProps<{
+  file: File
+  status: UploadStatus
+}>()
+
+const emit = defineEmits<{
+  (event: 'remove'): void
+}>()
+
+const { t } = useI18n()
+const { format } = useHumanFileSize()
+
+const thumbnail = URL.createObjectURL(file)
+onBeforeUnmount(() => URL.revokeObjectURL(thumbnail))
```

Comment:
maybe we should have a composable that abstracts the creation and deletion of the URL objects so that we never forget to revoke it.

Reply:
Clever!